src/jit/flowgraph.cpp

```diff
@@ -10466,10 +10466,8 @@ void Compiler::fgCompactBlocks(BasicBlock* block, BasicBlock* bNext)
             block->bbWeight = bNext->bbWeight;
 
             block->bbFlags |= (bNext->bbFlags & BBF_PROF_WEIGHT); // Set the profile weight flag (if necessary)
-            if (block->bbWeight != 0)
-            {
-                block->bbFlags &= ~BBF_RUN_RARELY; // Clear any RarelyRun flag
-            }
+            assert(block->bbWeight != 0);
```

Comment:
it if under `(block->bbWeight < bNext->bbWeight)`  and `bbWeight` is unsigned, so `bNext->bbWeight > 0` and we do ` block->bbWeight = bNext->bbWeight;`, so `block->bbWeight > 0`.

Reply:
I'm surprised that there wasn't a complaint about the line above that has `(block->bbWeight && bNext->bbWeight)` in which neither of those are booleans.

Reply:
I think VS has a 4 lvl warning about implicit casts from int to bool that is disabled in our source base, but it is a common thing.
We will be fixing these warnings for days if we enable them.